### PR TITLE
Remove metatrix in concept hierarchy - fixes #1752 - master

### LINKED ISSUE #1752: Concept hierarchy are not showing concepts that have CONCEPT_ANCESTOR records
This is likely due to an override from the [metatrix](https://github.com/OHDSI/Atlas/blob/master/js/const.js#L148) we should potentially deprecate that functionality and have the hierarchy use the min_levels_of_seperation with +1/-1 to show nearest parent/children concepts.

### PR DESCRIPTION
**Pull request by anthonysena**
Remove metatrix and reformat the concept hierarchy display as detailed in #1752

### CODE REVIEW

File: js/pages/concept-sets/concept-manager.html [hunk 1]
```diff
@@ -68,94 +68,105 @@
...
-                                                    { title: 'RC', data: 'RECORD_COUNT'},
-                                                    { title: 'DRC', data: 'DESCENDANT_RECORD_COUNT'},
-                                                    { title: 'Distance', data: function (d) { return String(Math.max.apply(Math, d.RELATIONSHIPS.map(function (o) {return o.RELATIONSHIP_DISTANCE;})))}},
-                                                    { title: 'Domain', data: 'DOMAIN_ID' },
-                                                    { title: 'Vocabulary', data: 'VOCABULARY_ID', width: '100px' }
-                                            ]
-                                    }
-                                }">
-                    </table>
+                <div id="wrapperChildren" data-bind="visible: $component.hierarchyPillMode() == 'all' || $component.hierarchyPillMode() == 'children'">
```
Comment (pavgra) [inline js/pages/concept-sets/concept-manager.html]: Suggestion: `['all', 'children'].includes($component.hierarchyPillMode())`

File: js/pages/concept-sets/concept-manager.html [hunk 2]
```diff
@@ -68,94 +68,105 @@
...
-                                                        { title: 'RC', data: 'RECORD_COUNT'},
-                                                        { title: 'DRC', data: 'DESCENDANT_RECORD_COUNT'},
-                                                        { title: 'Distance', data: function (d) { return String(Math.min.apply(Math, d.RELATIONSHIPS.map(function (o) {return o.RELATIONSHIP_DISTANCE;})))}},
-                                                        { title: 'Domain', data: 'DOMAIN_ID' },
-                                                        { title: 'Vocabulary', data: 'VOCABULARY_ID', width: '100px' }
-                                                ]
-                                        }
-                                    }">
-                    </table>
+                <div id="wrapperHierarchyAnchor" data-bind="visible: $component.hierarchyPillMode() == 'all' || $component.hierarchyPillMode() == 'current'">
```
Comment (pavgra) [inline js/pages/concept-sets/concept-manager.html]: Suggestion: `['all', 'current'].includes($component.hierarchyPillMode())`

File: js/pages/concept-sets/concept-manager.html [hunk 3]
```diff
@@ -68,94 +68,105 @@
...
+							<ul class="nav nav-pills">
+								<li role="presentation" data-bind="css: {active: $component.hierarchyPillMode() == 'all' }, click: function() {$component.hierarchyPillMode('all') };"><a>Full Hierarchy</a></li>
+								<li role="presentation" data-bind="css: {active: $component.hierarchyPillMode() == 'parents' }, click: function() {$component.hierarchyPillMode('parents') };"><a>Parents</a></li>
+								<li role="presentation" data-bind="css: {active: $component.hierarchyPillMode() == 'current' }, click: function() {$component.hierarchyPillMode('current') };"><a>Current</a></li>
+								<li role="presentation" data-bind="css: {active: $component.hierarchyPillMode() == 'children' }, click: function() {$component.hierarchyPillMode('children') };"><a>Children</a></li>
+							</ul>        
+						</div>
+					</div>
+				</div>
+                <div id="wrapperParents" data-bind="visible: $component.hierarchyPillMode() == 'all' || $component.hierarchyPillMode() == 'parents'">
```
Comment (pavgra) [inline js/pages/concept-sets/concept-manager.html]: Suggestion: `['all', 'parents'].includes($component.hierarchyPillMode())`

File: js/pages/concept-sets/concept-manager.js [hunk 4]
```diff
@@ -157,187 +159,31 @@ define([
...
+					'caption': 'Has Records',
+					'binding': function (o) {
+						return parseInt(o.RECORD_COUNT.toString()
+							.replace(',', '')) > 0;
+					}
+				}, {
+					'caption': 'Has Descendant Records',
+					'binding': function (o) {
+						return parseInt(o.DESCENDANT_RECORD_COUNT.toString()
+							.replace(',', '')) > 0;
```
[the remark this answers is not in the document]
Comment (anthonysena) [reply js/pages/concept-sets/concept-manager.js]: Seems to be a potential problem across the code base - I'll investigate and resolve via this PR.

File: js/pages/concept-sets/concept-manager.html (hunk 2 above)
Comment (anthonysena) [reply js/pages/concept-sets/concept-manager.html]: Tried this as: `visible: ['all', 'children'].includes($component.hierarchyPillMode())` as suggested but it appears the knockout binding does not respond to changes in the observable when it is nested inside of the `includes` function. So, I'll leave for now unless you have another suggestion on this one.

File: js/pages/concept-sets/concept-manager.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (anthonysena) [reply js/pages/concept-sets/concept-manager.js]: Resolved via changes here: https://github.com/OHDSI/Atlas/pull/1785/commits/ef9e8fd3d98d531b2b36c04f494f68d882b7ada3

**Comment by anthonysena** [comment]
@pavgra - I've updated the code based on the review (thank you!). As discussed, let's wait until we get feedback from @pbr6cornell regarding the hierarchy behavior before merging this work in case we need to make further adjustments. Thanks!

**Comment by pavgra** [comment]
@anthonysena , I would place just caption instead of the whole table & facets block when list of parents / children is empty. That would allow saving vertical space and, I believe, improves readibility
image

**Pull request merged**